esmvaltool/diag_scripts/ocean3d/interpolation.py

```diff
@@ -0,0 +1,248 @@
+"""APPLICATE/TRR Ocean Diagnostics."""
+import logging
+import os
+import ESMF
+import numpy as np
+import pyresample
+from cartopy.util import add_cyclic_point
+from netCDF4 import Dataset
+
+from esmvaltool.diag_scripts.ocean3d.getdata import load_meta
+
+logger = logging.getLogger(os.path.basename(__file__))
+
+
+def closest_depth(depths, depth):
+    """Find closest depth.
+
+    From vector of depths finds target depth,
+    that is closest to desired depth, Also returns
+    an index for the desired depth.
+    """
+    target_level = abs(abs(depths) - abs(depth)).argmin()
+    target_depth = depths[target_level]
+    logger.debug('target_depth: %s', target_depth)
+    return target_depth, target_level
+
+
+def interpolate_vert(depth_model, target_depth, data_model):
+    """Vertical linear interpolation.
```

Comment:
Vertical interpolation is available in the preprocessor. What is the advantage of this function?